src/pages/Map/components/LocatorControl.js

```diff
@@ -26,49 +29,75 @@ const LocatorButton = styled.button`
   }
 `;
 
-class LocatorControl extends PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isLoading: false
-    };
-  }
+const locateErrors = {
+  // https://developer.mozilla.org/en-US/docs/Web/API/GeolocationPositionError/code
+  PERMISSION_DENIED: 1,
+  POSITION_UNAVAILABLE: 2,
+  TIMEOUT: 3
+};
+
+const userFeedback = oneLine`Wenn Du Dich orten willst, musst Du einer Ortung zustimmen.
+  Du kannst Die Entscheidung, Deinen Standort nicht zu teilen, in den Einstellungen
+  Deines Browsers rückgängig machen.
+`;
 
-  locate = async () => {
-    this.setState({ isLoading: true });
-    this.props.onStart();
+const LocatorControl = ({ position, customPosition, onChange, onStart }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isGeolocationDenied, setIsGeolocationDenied] = useState(false);
+  const [isError, setIsError] = useState(false);
 
-    try {
-      const position = await getGeoLocation();
-      const lat = idx(position, (_) => _.coords.latitude);
-      const lng = idx(position, (_) => _.coords.longitude);
+  const onLocateSuccess = (geoPosition) => {
+    const lat = geoPosition?.coords?.latitude;
+    const lng = geoPosition?.coords?.longitude;
+    const isValidCoords = [lat, lng].every(isNumeric);
+    if (!isValidCoords) {
+      logger('invalid coords');
+    } else {
+      onChange([lng, lat]);
+    }
+  };
 
-      if (typeof lat === 'number' && typeof lng === 'number') {
-        this.props.onChange([lng, lat]);
-      }
-    } catch (err) {
-      alert('Um dich zu lokalisieren, benötigen wir deine Berechtigung.'); // eslint-disable-line
-      throw err;
+  const onLocateError = ({ code, message }) => {
+    logger(message);
+    if (code === locateErrors.PERMISSION_DENIED) {
+      setIsGeolocationDenied(true);
```

Comment:
Hier wird die Ablehnung der Ortung als Zustand der Komponente erfasst. Tatsächlich kann ich die Ablehnung aber außerhalb (in den Browsereinstellungen) ändern, ohne, dass der hier erfasst Zustand aktualisiert wird. 

Zum Beispiel: Ich klicke auf den Locator, der Browser fragt mich nach meiner Zustimmung, ich entscheide mich doch dagegen und lehne in der Browser-UI ab. Dann lese ich den Hinweis, dass deswegen keine Ortung möglich ist. Ich habe jetzt verstanden, warum ich der Abfrage vom Browser zustimmen sollte. Ich ändere meine Browser-Einstellung und klicke wieder auf den Locator. Jetzt ist allerdings der Button deaktiviert, da die Komponente erfasst hat: `isGeolocationDenied: true`. Ich kann den Locator nicht mehr klicken.

Stattdessen würde ich den State `isGeoLocationDenied` komplett rausnehmen. Dann kann ich in obigem Beispiel meine Browser-Einstellung aktualisieren und es ein zweites Mal versuchen, ohne die Seite ganz neu laden zu müssen.

Reply:
Das könnte man tun. Es ergibt sich dann jedoch ein anderer Nachteil:

Wenn der User nach der ersten Ablehnung *nicht* seine Browsereinstellungen entsprechend aktualisiert, führt ein Klick auf deb dann immernoch klickbaren Button dazu, dass die Fehlermeldung wieder erscheint. Probier mal.

Bin mir unsicher, was hier der bessere Weg ist.

Reply:
Denke jetzt Du hast Recht, das Property komplett rauszunehmen. Der Button ist dann nach zwar nach einem Verbot der Standortnutzung immernoch clickbar, zeigt ab er eine sinnvolle Meldung an. Das von Dir beschriebene Problem tritt dann nicht mehr auf.